### PR TITLE
Openstack progress

### PR DESCRIPTION
**Pull request by wester-is-not-real**
Added script that outputs the machines that are still running a python3.8 process. For this, I changed the run_cmds_on_machine function so it would output the response return code.

### CODE REVIEW

File: scripts/openstack/progress.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    openstack_dir = os.path.dirname(os.path.realpath(__file__))
+    # Load the machine IPs.
+    machine_file = os.path.join(openstack_dir, args.machines)
+    with open(machine_file, "r", encoding="utf-8") as f:
+        machines = f.read().splitlines()
+    # Make sure that the ssh key exists.
+    assert os.path.exists(args.sshkey)
+    # Loop through each machine and print those that have a python3.8 process.
+    progress_cmd = "pgrep python3.8"
+    print("The following machines are still running:\n")
```
Comment (tomsilver) [inline scripts/openstack/progress.py:34]: can we print something different if no machines are running?

**Review by tomsilver — COMMENTED** [review]
yay awesome, thanks for doing this 🙏

File: scripts/openstack/progress.py (hunk 1 above)
Comment (wester-is-not-real) [reply scripts/openstack/progress.py:34]: Great idea, on it.

File: scripts/openstack/progress.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
+"""Script for getting status of active openstack predicator experiments.
+
+WARNING: Will return the status of any other python3.8 processes running on
+        each machine in machines.txt
```
Comment (ronuchit) [inline scripts/openstack/progress.py:4]: nit: indent by one more space to align with previous line

File: scripts/openstack/progress.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    args = parser.parse_args()
+    openstack_dir = os.path.dirname(os.path.realpath(__file__))
+    # Load the machine IPs.
+    machine_file = os.path.join(openstack_dir, args.machines)
+    with open(machine_file, "r", encoding="utf-8") as f:
+        machines = f.read().splitlines()
+    # Make sure that the ssh key exists.
+    assert os.path.exists(args.sshkey)
+    # Loop through each machine and print those that have a python3.8 process.
+    progress_cmd = "pgrep python3.8"
```
Comment (ronuchit) [inline scripts/openstack/progress.py:33]: is there a reason you're only checking for python 3.8 in particular? what if someone is running 3.9 or 3.7? what would go wrong if you just do "pgrep python" here?

File: scripts/openstack/progress.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    print("The following machines are still running:\n")
+    for machine in machines:
+        # If return code is 0, print active machine.
+        returncode = run_cmds_on_machine([progress_cmd],
+                                         machine,
+                                         args.sshkey,
+                                         allowed_return_codes=(0, 1))
+
+        if returncode != 1:
+            print(f"{machine}")
```
Comment (ronuchit) [inline scripts/openstack/progress.py:43]: nit: you can just do `print(machine)`

File: scripts/openstack/progress.py (hunk 3 above)
Comment (wester-is-not-real) [reply scripts/openstack/progress.py:33]: launch.py specifically chooses python3.8 for the machines, so it shouldn't be an issue.

**Comment by tomsilver** [comment]
this command might work better: `ps -ef | grep python`

**Comment by tomsilver** [comment]
gonna close this for now and reopen on an as-needed basis

**Pull request closed without merging**